java/src/main/java/com/logicalclocks/hsfs/engine/SparkEngine.java

```diff
@@ -769,10 +769,23 @@ public <S> List<Feature> parseFeatureGroupSchema(S datasetGeneric,
     return features;
   }
 
-  public <S> S sanitizeFeatureNames(S datasetGeneric) {
+  public <S> S convertToDefaultDataframe(S datasetGeneric, List<String> nonNullColumns) {
     Dataset<Row> dataset = (Dataset<Row>) datasetGeneric;
-    return (S) dataset.select(Arrays.asList(dataset.columns()).stream().map(f -> col(f).alias(f.toLowerCase())).toArray(
-        Column[]::new));
+    Dataset<Row> sanitizedNamesDataset = dataset.select(Arrays.asList(dataset.columns()).stream().map(f ->
+            col(f).alias(f.toLowerCase())).toArray(Column[]::new));
+
+    if (nonNullColumns != null) {
+      Dataset<Row> datasetCopy = sanitizedNamesDataset.as("df1");
```

Comment:
not sure why this additional copy is needed?

Reply:
Indeed not needed